lib/grammers-client/src/client/messages.rs

```diff
@@ -546,6 +546,119 @@ impl Client {
         })
     }
 
+    /// Sends a album to the desired chat.
+    ///
+    /// This method can also be used to send a bunch of media such as photos, videos, documents, polls, etc.
+    ///
+    /// If you want to send a local file as media, you will need to use
+    /// [`Client::upload_file`] first.
+    ///
+    /// Refer to [`InputMedia`] to learn more formatting options, such as using markdown.
+    ///
+    /// See also: [`Message::respond_album`], [`Message::reply_album`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # async fn f(chat: grammers_client::types::Chat, client: grammers_client::Client) -> Result<(), Box<dyn std::error::Error>> {
+    /// use grammers_client::InputMedia;
+    ///
+    /// client.send_album(&chat, vec![InputMedia::caption("A album").photo_url("https://www.pixelstalk.net/wp-content/uploads/2016/03/Free-cute-animal-wallpaper-HD.jpg")]).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// [`InputMedia`]: crate::InputMedia
+    pub async fn send_album<C: Into<PackedChat>>(
+        &self,
+        chat: C,
+        mut medias: Vec<InputMedia>,
+    ) -> Result<Vec<Message>, InvocationError> {
+        let chat = chat.into();
+        let random_ids = generate_random_ids(medias.len());
+
+        // Upload external files
+        for media in medias.iter_mut() {
+            let raw_media = media.media.clone().unwrap();
+
+            if matches!(
+                raw_media,
+                tl::enums::InputMedia::PhotoExternal(_)
+                    | tl::enums::InputMedia::DocumentExternal(_)
+            ) {
+                let uploaded = self
+                    .invoke(&tl::functions::messages::UploadMedia {
+                        business_connection_id: None,
+                        peer: chat.to_input_peer(),
+                        media: raw_media,
+                    })
+                    .await?;
+                media.media = Some(
+                    types::Media::from_raw(uploaded)
+                        .unwrap()
+                        .to_raw_input_media()
+                        .unwrap(),
+                );
+            }
+        }
+
+        let first_media = medias.first().unwrap();
```

Comment:
Oh yeah, I hadn't thought about attributes that only make sense for the entire group. This seems unfortunate. But might be a fair compromise.

Reply:
indeed, it's unfornate